api/request_source/request_source_plugin.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+package nighthawk.request_source;
+
+import "google/protobuf/wrappers.proto";
+import "validate/validate.proto";
+
+// Configuration for DummyPluginRequestSource (plugin name: "nighthawk.dummy-request-source-plugin")
+// that does nothing
+message DummyPluginRequestSourceConfig {
+  // Dummy input value.
+  google.protobuf.DoubleValue dummy_value = 1;
+}
+
+// Configuration for RPCPluginRequestSource (plugin name: "nighthawk.rpc-request-source-plugin")
+// that uses rpc
+message RPCPluginRequestSourceConfig {
+  string uri = 1;
+}
+
+// Configuration for FileBasedPluginRequestSource (plugin name:
+// "nighthawk.file-based-request-source-plugin") that uses rpc
+message FileBasedPluginRequestSourceConfig {
+  string uri = 1;
+  string file_path = 2;
+  google.protobuf.UInt32Value num_requests = 3 [(validate.rules).uint32 = {gte: 0, lte: 1000000}];
+  google.protobuf.UInt32Value max_file_size = 4;
```

Comment:
Naive question: we'll be reading a proto stream from disk, right? I have no idea how that works, but that won't be subject to things along the lines of [zip boms](https://en.wikipedia.org/wiki/Zip_bomb) right?